core/src/main/java/com/linecorp/armeria/client/HttpClientDelegate.java

```diff
@@ -115,14 +121,31 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Ex
                     acquireConnectionAndExecute(ctx, resolved, req, res, timingsBuilder, proxyConfig);
                 } else {
                     ctx.logBuilder().session(null, ctx.sessionProtocol(), timingsBuilder.build());
-                    earlyFailedResponse(cause, ctx, req, res);
+                    ctx.cancel(cause);
                 }
             });
         }
 
         return res;
     }
 
+    private static void updateCancellationTask(ClientRequestContext ctx, HttpRequest req,
+                                               DecodedHttpResponse res) {
+        final ClientRequestContextExtension ctxExt = ctx.as(ClientRequestContextExtension.class);
+        if (ctxExt == null) {
+            return;
+        }
+        ctxExt.responseCancellationScheduler().updateTask(cause -> {
```

Comment:
I suggested removing `updateTask()` method, but that would result in too many changes to the existing code. As long as the implemetation is correct, I'm okay to maintain the current style and focus on implementing the response timeout mode.

Reply:
For the record, `updateTask()` doesn't necessarily have to be removed - it can still be used internally since the old `CancellationScheduler#start` acted similarly to `updateTask()`.

It is probably more important to unify and simplify the request/response abortion mechanism. This seems difficult to do at the moment for client-side. In particular, when one attempts this one should take note of 1) consistency between `[req|res].cause` and `log.[req|res]Cause`, 2) setting the correct exception type (e.g. `UnprocessedRequestException`).